Task20 design patterns

andresmadeddie opened the pull request:
Create a Builder folder with builder classes to create objects from the model folder. Builder Design Patters

andresmadeddie committed:
Add folder Builder with an implementation of builder pattern for models

andresmadeddie committed:
implement builders in main class

ankochergin commented on src/main/java/com/solvd/sql/builder/CategoryBuilder.java:
```diff
@@ -0,0 +1,32 @@
+        category.setCategoryName(this.categoryName);
+        return category;
+    }
+}
```
Usually builder class is an inner class of a model. Here I suggest to:
1. make CategoryBuilder class as an inner class of Category. 
2.  make your class static and remove builder () method. so you'll be able to use it like Category.CategoryBuilder().withId(...).withCategoryName(...).build();
3. to Category class add constructor with builder as a parameter
your build() method will look like:
```
public Category build() {
        return new Category(this);
    }
```

andresmadeddie committed:
remove folder Builders with all its content
Modify Category and Customer classes adding builder inner class and modifying dao files and main class accordingly

andresmadeddie committed:
Person
modify classes adding builder inner class and modifying dao files and main class accordingly

andresmadeddie committed:
Person, Shop
modify classes adding builder inner class and modifying dao files and main class ,and xmlparser file accordingly

andresmadeddie committed:
Order, OrderItem, Staff, Product, Supplier
modify classes adding builder inner class and modifying dao files and main class ,and xmlparser file accordingly

andresmadeddie committed:
Promotion, ProductPromotion,
modify classes adding builder inner class and modifying dao files and main class.

ankochergin commented on src/main/java/com/solvd/Main.java:
```diff
@@ -51,35 +45,14 @@ public static void main(String[] args) throws JAXBException {
+        personService.insert(new Person.Builder().withPersonName("Tom").withLastName("Sawyer").withPhone("1234561236").withAddress("756 nowhere St. Petersburg Missouri").build());
+        personService.insert(new Person.Builder().withPersonName("Emma").withLastName("Watson").withPhone("1234561237").withAddress("342 somewhere Liverpool, England").build());
+        personService.insert(new Person.Builder().withPersonName("Charlie").withLastName("Garcia").withPhone("1234561234").withAddress("234 brown street Los Angeles,CA").build());
+        personService.insert(new Person.Builder().withPersonName("Genghis").withLastName("Khan").withPhone("8569751246").withAddress("9832 cest' la vie Monroe, 239843").build());
```
please fix formatting, lines shouldn't be too long

ankochergin commented on src/main/java/com/solvd/sql/jdbc/CategoryDao.java:
```diff
@@ -83,10 +83,10 @@ public List<Category> getAll() {
-                    category.setId(rs.getInt("id"));
-                    category.setCategoryName(rs.getString("category_name"));
-                    categories.add(category);
+                    categories.add(new Category.Builder()
+                            .withId(rs.getInt("id")).
+                            withCategoryName(rs.getString("category_name"))
+                            .build());
```
```suggestion
                    categories.add(new Category.Builder()
                            .withId(rs.getInt("id"))
                            .withCategoryName(rs.getString("category_name"))
                            .build());
```

ankochergin replied:
the same mistake I see in other places

ankochergin commented on src/main/java/com/solvd/util/XmlParser.java:
```diff
@@ -20,16 +21,24 @@ public static Shop parseShopDataFromFile(String xmlFilePath) {
+                            .withId(Integer.parseInt(ownerElement.getElementsByTagName("person_id").item(0).getTextContent()))
+                            .build())
+                    .build();
+            shop.setOwner(owner);
```
formatting

andresmadeddie committed:
Main, CategoryDao, and XmlParser format corrections
Create line breaks, fixed and changed points formatting in the Main and CategoryDao classes. Likewise, checked all Dao Folder's files for similar formatting issues.
Create line breaks XmlParser.java

andresmadeddie committed:
Main, CategoryDao, and XmlParser format corrections
Create line breaks, fixed and changed points formatting in the Main and CategoryDao classes. Likewise, checked all Dao Folder's files for similar formatting issues.
Create line breaks XmlParser.java

andresmadeddie committed:
Merge remote-tracking branch 'origin/task20-DesignPatterns' into task20-DesignPatterns
